### PR TITLE
Base plugin schema update.

### PR DESCRIPTION
**Pull request by peterlogo**
### What this PR does (required):
- A base plugin for creating other card plugins.
- Instance methods were also added to the base plugin schema for attaching and detaching a plugin from a card.

### Any information needed to test this feature (required):
- This is basically the schema implementation which is needed for other plugins to be created.
- To test the full capability of the base plugin we would need a plugin that extends from the base plugin.

### CODE REVIEW

**Comment by refugedesigns** [comment]
Great job Boss!

File: server/app.js [hunk 1]
```diff
@@ -60,6 +60,10 @@ if (process.env.NODE_ENV === "production") {
 app.use(notFound);
 app.use(errorHandler);
 
+server.listen(8000, () => {
+  console.log("server running on port: 8000");
+});
+
```
Comment (moffatethan) [inline server/app.js:66]: We have a `bin/www` file for running our server. We don't need this

**Review by moffatethan — CHANGES_REQUESTED** [review]
One fix then I'll approve. Great work here Peter!

File: server/app.js (hunk 1 above)
Comment (peterlogo) [reply server/app.js:66]: ok @moffatethan let me take that line out.

**Review by moffatethan — APPROVED** [review]
Good work!

**Pull request merged**